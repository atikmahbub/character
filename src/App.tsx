import * as React from "react";
import Container from "@mui/material/Container";
import { routes } from "./routes/routes";
import { Routes, Route } from "react-router-dom";
import { styled } from "@mui/system";

const AppLayout = styled("div")({
  backgroundColor: "#cfe8fc",
});

const StyledContainer = styled(Container)({
  backgroundColor: "#fff",
  minHeight: "100vh",
  padding: 5,
});

export default function App() {
  return (
    <AppLayout>
      <StyledContainer maxWidth="lg">
        <React.Suspense fallback={<div>Loading...</div>}>
          <Routes>
            {routes.map((item: any) => (
              <Route
                key={item.name}
                path={item.pathname}
                element={<item.component />}
              />
            ))}
          </Routes>
        </React.Suspense>
      </StyledContainer>
    </AppLayout>
  );
}
