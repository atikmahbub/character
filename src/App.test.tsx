import React from "react";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import App from "./App";

const jsx = (
  <BrowserRouter>
    <App />
  </BrowserRouter>
);

test("Render Without Crashing Main App", () => {
  const tree = renderer.create(jsx).toJSON();
  expect(tree).toMatchSnapshot();
});
