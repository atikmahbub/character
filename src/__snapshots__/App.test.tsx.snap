// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Without Crashing Main App 1`] = `
<div
  className="css-1g7p4m1"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1ktt8pt-MuiContainer-root"
  >
    <div
      className="css-1cegspo"
    >
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorSecondary css-1szo49z-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": 40,
            "width": 40,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
